app/src/app/components/specmath-modal/modal.component.spec.ts

```diff
@@ -80,49 +76,30 @@ describe('ModalComponent', () => {
     expect(spy).toHaveBeenCalled();
   });
 
-  it('next button is disabled on step 1 when the new file name input is empty', () => {
+  it('next button is disabled', () => {
     fixture.detectChanges();
-    const nextButton = queryElement(fixture, '#modal-step-1-next').nativeElement;
+    const nextButton = queryElement(fixture, '#modal-button-next').nativeElement;
 
     expect(nextButton.disabled).toBeTruthy();
   });
 
-  it('new file name form validity', () => {
+  it('next button is enabled when newFileName is valid', () => {
     fixture.detectChanges();
-    const specNameField = modal.specNameFormControl;
-    expect(specNameField.valid).toBeFalsy();
-
-    specNameField.setValue('invalid input!');
-    expect(specNameField.hasError('pattern')).toBeTruthy();
-
-    specNameField.setValue('');
-    expect(specNameField.hasError('required')).toBeTruthy();
+    const nextButton = queryElement(fixture, '#modal-button-next').nativeElement;
 
-    specNameField.setValue('test_spec');
-    expect(specNameField.valid).toBeTruthy();
-  });
-
-  it('calls nextStep() when the new file name is valid and the next button is clicked', () => {
+    modal.step1Options = { newFileName: 'new_spec', valid: true };
```

Comment:
... the idea being you emit (options) from the stub.  This gives you less reliance on the subject component's "internals", and it exercises/tests the wire-up of inputs and outputs between the subject and the subcomponent.
I can give you some more detailed examples.

Reply:
Done 👍